CMakeLists.txt

```diff
@@ -372,10 +372,10 @@ function(add_iree_mlir_src_dep llvm_monorepo_path)
   set(CMAKE_BUILD_TYPE "${_CMAKE_BUILD_TYPE}" CACHE STRING "Build type (default ${DEFAULT_CMAKE_BUILD_TYPE})" FORCE)
 endfunction()
 
-if(${IREE_ENABLE_MLIR})
-  if(${IREE_MLIR_DEP_MODE} STREQUAL "DISABLED")
-    message(STATUS "Not adding MLIR/LLVM dep due to IREE_MLIR_DEP_MODE=DISABLED")
-  elseif(${IREE_MLIR_DEP_MODE} STREQUAL "BUNDLED")
+if(NOT IREE_ENABLE_MLIR OR "${IREE_MLIR_DEP_MODE}" STREQUAL "DISABLED")
```

Comment:
we were wondering the other day if we needed the IREE_ENABLE_MLIR flag at all and could instead just use the IREE_ENABLE_COMPILER flag: can't build the compiler without mlir, and no reason to ever build mlir without the compiler using IREE's build system.

Reply:
Currently if IREE_ENABLE_COMPILER is OFF, then you don't process those directories at all. I do want those targets available, but I also want to tweak my own LLVM. A way to go on this would be to detect whether the LLVM/MLIR dep has been externally provided (i.e. `if(NOT TARGET MLIRIR)`). I'm running into this in all of the projects while unbundling dep finding and would like a simple heuristic/way.

Reply:
Sync'd offline - will switch this to feature detection instead of a flag. Will also hoist the LLVM setting from iree_copts.cmake into a more serviceable place.